src/react/useMembers.ts

```diff
@@ -0,0 +1,121 @@
+import { useEffect, useRef, useState } from 'react';
+import { useSpace } from './useSpace.js';
+import { isArray, isFunction, isString } from '../utilities/is.js';
+
+import type { ErrorInfo } from 'ably';
+import type { Space, SpaceMember } from '..';
+import type { UseSpaceOptions } from './types.js';
+
+interface UseMembersResult {
+  space?: Space;
+  /**
+   * All members present in the space
+   */
+  members: SpaceMember[];
+  /**
+   * All members present in the space excluding the member associated with the spaces client
+   */
+  others: SpaceMember[];
+  /**
+   * The member associated with the spaces client
+   */
+  self: SpaceMember | null;
+  channelError: ErrorInfo | null;
+  connectionError: ErrorInfo | null;
+}
+
+type UseMembersCallback = (params: SpaceMember) => void;
+
+type MembersEvent = 'leave' | 'enter' | 'update' | 'updateProfile' | 'remove';
```

Comment:
Events are already defined here https://github.com/ably/spaces/blob/main/src/Members.ts#L13, it would be nice to use them (they also define the argument which provides a better TS experience)